app/src/main/java/com/github/libretube/ui/sheets/ChannelGroupsSheet.kt

```diff
@@ -46,6 +49,35 @@ class ChannelGroupsSheet(
             dismiss()
         }
 
+        val callback = object : ItemTouchHelper.SimpleCallback(
+            ItemTouchHelper.UP or ItemTouchHelper.DOWN,
+            0
+        ) {
+            override fun onMove(
+                recyclerView: RecyclerView,
+                viewHolder: RecyclerView.ViewHolder,
+                target: RecyclerView.ViewHolder
+            ): Boolean {
+                val from = viewHolder.absoluteAdapterPosition
+                val to = target.absoluteAdapterPosition
+
+                groups.move(from, to)
+                adapter.notifyItemMoved(from, to)
+
+                groups.mapIndexed { index, subscriptionGroup -> subscriptionGroup.index = index }
```

Comment:
```suggestion
                groups.forEachIndexed { index, subscriptionGroup -> subscriptionGroup.index = index }
```

Reply:
I'm not familiar with Kotlin, what's the difference between the two? The documentation wasn't much help.

Reply:
`mapIndexed` creates a new list with the content specified in the lambda, while `forEachIndexed` just iterates over the items in the list.